src/main/scala/scalaz/analytics/AnalyticsModule.scala

```diff
@@ -100,19 +100,33 @@ trait AnalyticsModule {
    * The DataSet/DataStream operations of scalaz-analytics
    */
   trait Ops[F[_]] {
-    // Unbounded
+    // Unbounded dataset ops
+    // (These ops require only bounded space, even with unbounded data)
     def map[A, B](ds: F[A])(f: A =>: B): F[B]
+    def flatMap[A, B](ds: F[A])(f: A =>: DataSet[B]): F[B]
     def filter[A](ds: F[A])(f: A =>: Boolean): F[A]
+    // Streaming aggregations
+    // (These ops produce cumulative results, some may require unbounded space)
+    def scan[A, B](ds: F[A])(initial: Unit =>: B)(f: (B, A) =>: B): F[B]
 
-    // Bounded
-    def fold[A, B](ds: F[A])(window: Window)(initial: A =>: B)(f: (B, A) =>: B): F[B]
+    def scanAggregateBy[A, K, V](ds: F[A])(g: A =>: K)(initial: Unit =>: V)(
+      f: (V, A) =>: V
+    ): F[(K, V)]
+
+    // Bounded dataset ops
+    // (These ops can only be performed on bounded subsets of data)
+    def fold[A, B](ds: F[A])(window: Window)(initial: Unit =>: B)(f: (B, A) =>: B): F[B]
+
+    def aggregateBy[A, K, V](ds: F[A])(window: Window)(g: A =>: K)(initial: Unit =>: V)(
+      f: (V, A) =>: V
+    ): F[(K, V)]
     def distinct[A](ds: F[A])(window: Window): F[A]
```

Comment:
A cumulative/streaming variant of this operation could also be provided. Like scanAggregateBy, it would in general require unbounded space (i.e., space required would depend on the number of distinct elements).